Corrige code-analysis.

idgserpro opened the pull request:
Fecha assim https://github.com/plonegovbr/brasil.gov.portlets/issues/6

Inicia também o processo de tentar corrigir os testes prevendo release
1.1.4 de brasil.gov.portal.

@hvelarde p/c

idgserpro committed:
Corrige code-analysis.

Fecha assim https://github.com/plonegovbr/brasil.gov.portlets/issues/6

Inicia também o processo de tentar corrigir os testes prevendo release
1.1.4 de brasil.gov.portal.

hvelarde commented:
eu acho mais importante corregir os testes que estão falhando em outro branch antes de mesclar isso aqui. correger D001 é simples; favor incluir isso também.

idgserpro commented:
Os testes passam se removermos a pinagem que o IDG fez em plone.app.collection: se despinarmos a branch 2.x e usarmos a definida em https://github.com/plone/Installers-UnifiedInstaller/blob/4.3.9-r1/base_skeleton/versions.cfg#L138 passa a funcionar. [Criei uma branch](https://github.com/plonegovbr/brasil.gov.portlets/tree/issues_3_6_plone_app_collection_1_0_13) a partir dessa alterando essa versão pra mostrar os testes passando (veja em https://travis-ci.org/plonegovbr/brasil.gov.portlets/builds/136169308).

Não sei o motivo dessa pinagem e portanto não sei se podemos remover. @rodfersou @ericof alguma idéia? Por que se pinou plone.app.collection > 2.x no IDG?

idgserpro commented:
@hvelarde @rodfersou @ericof

Pegando o histórico do IDG (início de 2013) temos que o release de plone.app.contenttypes 1.0b2 ainda pede o plone.app.contenttypes 2.x para poder fornecer o tipo coleção como Dexterity (baixe o pacote no pypi e leia o seu README).

> It contains the same types as default Plone does:
> - Folder
> - Document
> - News item
> - File
> - Image
> - Link
> - Event (this will be replaced by plone.app.event soon)
> - **Collection (this is provided by plone.app.collection)**

A partir do plone.app.contenttypes 1.0rc1, plone.app.contenttypes versão 2.x não é mais necessário:

> It contains the same types as default Plone does:
> - Folder
> - Document
> - News item
> - File
> - Image
> - Link
> - Event (this will be replaced by plone.app.event soon)
> - **Collection (this already replaces plone.app.collection which is no longer needed then)**

Acreditamos que durante o processo de desenvolvimento foi necessário utilizar sim essa branch 2.x, mas aí quando saiu o release 1.0 de plone.app.contenttypes (que torna plone.app.collection desnecessário) e sua [versão pinada no release 1.0 do IDG](https://github.com/plonegovbr/portalpadrao.release/blob/ef862e9dae5ab44054c217a3c9a8e24747d9239f/1.0/versions.cfg#L481), foi esquecido de despinar o plone.app.collection. Queremos remover a pinagem do plone.app.collection para esse release 1.1.4, por favor dêem suas opiniões.

hvelarde commented:
@idgserpro eu não tenho recursos para trabalhar nisso; por favor implementem as mudanças que precisarem e me avisem quando ficar pronto para fazer o release.

idgserpro committed:
Remove o pending do release 1.1.4

Com o commit

    https://github.com/plonegovbr/portalpadrao.release/commit/1ee798ba9af481370644ccef61f432b44e0b91b5

o 'pending' do diretório do release foi removido.

idgserpro commented:
Pode efetuar o merge do PR.
